src/PactNet.Native/NativeMockServer.cs

```diff
@@ -168,15 +233,30 @@ private static class Interop
             [DllImport(dllName, EntryPoint = "with_body")]
             public static extern bool WithBody(InteractionHandle interaction, InteractionPart part, string contentType, string body);
 
+            [DllImport(dllName, EntryPoint = "message_with_metadata")]
+            public static extern bool MessageWithMetadata(MessageHandle message, string key, string value);
+
+            [DllImport(dllName, EntryPoint = "message_with_contents")]
+            public static extern bool MessageWithContents(MessageHandle message, string contentType, string body, UIntPtr size);
+
+            [DllImport(dllName, EntryPoint = "message_reify")]
+            public static extern IntPtr MessageReify(MessageHandle message);
+
             [DllImport(dllName, EntryPoint = "free_string")]
             public static extern void FreeString(IntPtr s);
         }
     }
 
     [StructLayout(LayoutKind.Sequential)]
-    internal readonly struct PactHandle
+    internal struct PactHandle
```

Comment:
These should be `readonly` - it's dangerous to change them after you've created them when doing the interop.